<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Smart-seq Example2</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Single Cell Clustering</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="large.html">Drop-seq / 10X Pipeline</a>
</li>
<li>
  <a href="small.html">High Coverage Pipeline</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tk382/SCNoisyClustering">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Smart-seq Example2</h1>

</div>


<p><strong>Last updated:</strong> 2018-09-14</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180618)</code> </summary></p>
<p>The command <code>set.seed(20180618)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/tk382/SCNoisyClustering/tree/86bd1c0f000a21c6808555e251878756a7afcc5b" target="_blank">86bd1c0</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    R/.Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/pipeline/.Rhistory

Untracked files:
    Untracked:  ..gif
    Untracked:  .DS_Store
    Untracked:  R/.DS_Store
    Untracked:  analysis/.DS_Store
    Untracked:  analysis/normalization_test.R
    Untracked:  analysis/pipeline/0_dropseq/
    Untracked:  analysis/pipeline/1_10X/
    Untracked:  analysis/pipeline/2_zeisel/
    Untracked:  analysis/pipeline/3_smallsets/
    Untracked:  analysis/slsl_10x.Rdata
    Untracked:  analysis/slsl_dropseq.Rdata
    Untracked:  analysis/writeup/bibliography.bib
    Untracked:  analysis/writeup/draft1.aux
    Untracked:  analysis/writeup/draft1.bbl
    Untracked:  analysis/writeup/draft1.blg
    Untracked:  analysis/writeup/draft1.log
    Untracked:  analysis/writeup/draft1.out
    Untracked:  analysis/writeup/draft1.pdf
    Untracked:  analysis/writeup/draft1.synctex.gz
    Untracked:  analysis/writeup/draft1.tex
    Untracked:  analysis/writeup/jabbrv-ltwa-all.ldf
    Untracked:  analysis/writeup/jabbrv-ltwa-en.ldf
    Untracked:  analysis/writeup/jabbrv.sty
    Untracked:  analysis/writeup/naturemag-doi.bst
    Untracked:  analysis/writeup/wlscirep.cls
    Untracked:  data/unnecessary_in_building/
    Untracked:  docs/figure/10x_labeled.Rmd/
    Untracked:  docs/figure/example_10x.Rmd/.DS_Store
    Untracked:  docs/figure/large.Rmd/
    Untracked:  dropseq_heatmap.pdf
    Untracked:  man/correct_detection_rate.Rd

Unstaged changes:
    Modified:   NAMESPACE
    Modified:   R/SLSL.R
    Modified:   R/correct_detection_rate.R
    Modified:   R/find_markers.R
    Modified:   analysis/pipeline/.DS_Store
    Modified:   analysis/writeup/.DS_Store
    Modified:   data/.DS_Store
    Modified:   docs/figure/.DS_Store
    Deleted:    man/correct.detection.rate.Rd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/86bd1c0f000a21c6808555e251878756a7afcc5b/analysis/small.Rmd" target="_blank">86bd1c0</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-09-14
</td>
<td style="text-align:left;">
wflow_publish(c(“analysis/_site.yml“,”analysis/large.Rmd“,”analysis/small.Rmd“), republish
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="read-data-yan" class="section level2">
<h2>Read Data : Yan</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&#39;data/Yan.rda&#39;</span>)
X         =<span class="st"> </span><span class="kw">as.matrix</span>(yan)
genenames =<span class="st"> </span><span class="kw">rownames</span>(X)
truelabel =<span class="st"> </span><span class="kw">as.numeric</span>(<span class="kw">as.factor</span>(ann<span class="op">$</span>cell_type1))
numClust =<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(truelabel))</code></pre></div>
</div>
<div id="quality-control-and-cell-filtering" class="section level2">
<h2>Quality Control and Cell Filtering</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summary =<span class="st"> </span><span class="kw">explore_data</span>(X, genenames)</code></pre></div>
<p><img src="figure/small.Rmd/cell_filter-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-filtering" class="section level2">
<h2>Gene Filtering</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">disp =<span class="st"> </span><span class="kw">plot_dispersion</span>(<span class="dt">X =</span> X, 
                       <span class="dt">genenames =</span> genenames, 
                       <span class="dt">bins=</span><span class="ot">NA</span>, 
                       <span class="dt">median =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="figure/small.Rmd/gene_filter-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">X =<span class="st"> </span><span class="kw">gene_filter</span>(X, genenames, disp, 
                <span class="dt">mean.thresh=</span><span class="kw">c</span>(<span class="op">-</span><span class="ot">Inf</span>, <span class="dv">4000</span>), 
                <span class="dt">dispersion.thresh =</span> <span class="kw">c</span>(<span class="fl">0.5</span>, <span class="ot">Inf</span>))

genenames =<span class="st"> </span>X<span class="op">$</span>genenames
X =<span class="st"> </span>X<span class="op">$</span>X</code></pre></div>
</div>
<div id="normalization-and-bias-correction" class="section level2">
<h2>Normalization and Bias Correction</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">X =<span class="st"> </span><span class="kw">quantile_normalize</span>(X)
cd =<span class="st"> </span><span class="kw">correct_detection_rate</span>(<span class="kw">as.matrix</span>(X), <span class="dt">det.rate =</span> <span class="kw">colMeans</span>(X<span class="op">&gt;</span><span class="dv">0</span>)<span class="op">/</span><span class="kw">nrow</span>(X))</code></pre></div>
<p><img src="figure/small.Rmd/normalize_and_correction-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="run-the-algorithm" class="section level2">
<h2>Run the algorithm</h2>
<p>Run SLSL on the log.cpm matrix.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">out =<span class="st"> </span><span class="kw">SLSL</span>(X, <span class="dt">numClust =</span> numClust)</code></pre></div>
<p><img src="figure/small.Rmd/slsl-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">adj.rand.index</span>(out<span class="op">$</span>result, truelabel)</code></pre></div>
<pre><code>[1] 0.8954618</code></pre>
</div>
<div id="analyze-differentially-expressed-genes" class="section level2">
<h2>Analyze : Differentially Expressed Genes</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">degenes =<span class="st"> </span><span class="kw">de_genes</span>(X, genenames, out<span class="op">$</span>result, <span class="dt">top.n=</span><span class="dv">100</span>, <span class="dt">plot=</span><span class="dv">6</span>)</code></pre></div>
<p><img src="figure/small.Rmd/de-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(degenes)</code></pre></div>
<pre><code>  de_genes   log10p
1    HDAC9 16.36138
2   STK32B 16.28383
3 FLJ40852 16.19759
4    KPNA7 16.10733
5   MFSD2A 16.10030
6     WEE2 16.02111</code></pre>
</div>
<div id="markers-for-each-cluster" class="section level2">
<h2>Markers for Each Cluster</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bio.markers =<span class="st"> </span><span class="kw">find_markers</span>(X, genenames, out<span class="op">$</span>result, out<span class="op">$</span>tsne<span class="op">$</span>Y, <span class="dt">top.n =</span> <span class="dv">50</span>, <span class="dt">plot.n=</span><span class="dv">3</span>)
bio.markers<span class="op">$</span>plots[[<span class="dv">1</span>]]</code></pre></div>
<p><img src="figure/small.Rmd/bio-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(bio.markers<span class="op">$</span>markers[[<span class="dv">1</span>]])</code></pre></div>
<pre><code>  clust1_genenames clust1_log10p
1             PDK4      17.15950
2          NEUROD1      15.94980
3         C11orf96      13.93428
4         FLJ37201      12.38675
5           HOXD11      12.18412
6           PCDH10      11.56774</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bio.markers<span class="op">$</span>plots[[<span class="dv">2</span>]]</code></pre></div>
<p><img src="figure/small.Rmd/bio-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(bio.markers<span class="op">$</span>markers[[<span class="dv">2</span>]])</code></pre></div>
<pre><code>  clust2_genenames clust2_log10p
1         C19orf33      19.08016
2        C10orf125      18.53046
3            KRT19      18.03959
4            CRIP1      18.02379
5         C9orf140      17.76193
6             BIN1      17.53935</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bio.markers<span class="op">$</span>plots[[<span class="dv">3</span>]]</code></pre></div>
<p><img src="figure/small.Rmd/bio-3.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(bio.markers<span class="op">$</span>markers[[<span class="dv">3</span>]])</code></pre></div>
<pre><code>  clust3_genenames clust3_log10p
1           DUX4L2      12.67983
2             DUX4      11.99765
3           DUX4L3      11.13193
4          USP17L2      10.71518
5           DUX4L5      10.49148
6     LOC100127888      10.40254</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.5

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] bindrcpp_0.2.2          gridExtra_2.3          
 [3] gdata_2.18.0            stargazer_5.2.2        
 [5] abind_1.4-5             broom_0.5.0            
 [7] gplots_3.0.1            diceR_0.5.1            
 [9] Rtsne_0.13              igraph_1.2.2           
[11] scatterplot3d_0.3-41    pracma_2.1.4           
[13] fossil_0.3.7            shapefiles_0.7         
[15] foreign_0.8-71          maps_3.3.0             
[17] sp_1.3-1                caret_6.0-80           
[19] lattice_0.20-35         reshape_0.8.7          
[21] dplyr_0.7.6             ggplot2_3.0.0          
[23] irlba_2.3.2             Matrix_1.2-14          
[25] quadprog_1.5-5          inline_0.3.15          
[27] matrixStats_0.54.0      SCNoisyClustering_0.1.0

loaded via a namespace (and not attached):
 [1] nlme_3.1-137              bitops_1.0-6             
 [3] lubridate_1.7.4           dimRed_0.1.0             
 [5] rprojroot_1.3-2           tools_3.5.1              
 [7] backports_1.1.2           R6_2.2.2                 
 [9] KernSmooth_2.23-15        rpart_4.1-13             
[11] lazyeval_0.2.1            colorspace_1.3-2         
[13] nnet_7.3-12               withr_2.1.2              
[15] tidyselect_0.2.4          compiler_3.5.1           
[17] git2r_0.23.0              labeling_0.3             
[19] caTools_1.17.1.1          scales_0.5.0             
[21] sfsmisc_1.1-2             DEoptimR_1.0-8           
[23] robustbase_0.93-2         stringr_1.3.1            
[25] digest_0.6.15             rmarkdown_1.10           
[27] R.utils_2.6.0             pkgconfig_2.0.1          
[29] htmltools_0.3.6           rlang_0.2.1              
[31] ddalpha_1.3.4             bindr_0.1.1              
[33] gtools_3.8.1              mclust_5.4.1             
[35] ModelMetrics_1.1.0        R.oo_1.22.0              
[37] magrittr_1.5              Rcpp_0.12.18             
[39] munsell_0.5.0             R.methodsS3_1.7.1        
[41] stringi_1.2.4             whisker_0.3-2            
[43] yaml_2.2.0                MASS_7.3-50              
[45] plyr_1.8.4                recipes_0.1.3            
[47] grid_3.5.1                pls_2.6-0                
[49] crayon_1.3.4              splines_3.5.1            
[51] knitr_1.20                pillar_1.3.0             
[53] reshape2_1.4.3            codetools_0.2-15         
[55] stats4_3.5.1              CVST_0.2-2               
[57] magic_1.5-8               glue_1.3.0               
[59] evaluate_0.11             RcppArmadillo_0.8.600.0.0
[61] data.table_1.11.4         foreach_1.4.4            
[63] gtable_0.2.0              purrr_0.2.5              
[65] tidyr_0.8.1               kernlab_0.9-26           
[67] assertthat_0.2.0          DRR_0.0.3                
[69] gower_0.1.2               prodlim_2018.04.18       
[71] class_7.3-14              survival_2.42-6          
[73] geometry_0.3-6            timeDate_3043.102        
[75] RcppRoll_0.3.0            tibble_1.4.2             
[77] iterators_1.0.10          workflowr_1.1.1          
[79] lava_1.6.2                ipred_0.9-6              </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
