<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>How to Use SLSL, Case by Case</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code > span.dt { color: #204a87; } /* DataType */
code > span.dv { color: #0000cf; } /* DecVal */
code > span.bn { color: #0000cf; } /* BaseN */
code > span.fl { color: #0000cf; } /* Float */
code > span.ch { color: #4e9a06; } /* Char */
code > span.st { color: #4e9a06; } /* String */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #8f5902; } /* Other */
code > span.al { color: #ef2929; } /* Alert */
code > span.fu { color: #000000; } /* Function */
code > span.er { color: #a40000; font-weight: bold; } /* Error */
code > span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #000000; } /* Constant */
code > span.sc { color: #000000; } /* SpecialChar */
code > span.vs { color: #4e9a06; } /* VerbatimString */
code > span.ss { color: #4e9a06; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #000000; } /* Variable */
code > span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code > span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code > span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code > span.ex { } /* Extension */
code > span.at { color: #c4a000; } /* Attribute */
code > span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code > span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Single Cell Clustering</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="tutorial.html">Tutorial</a>
</li>
<li>
  <a href="detection_rate_correction.html">Detection Rate Correction</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tk382/SCNoisyClustering">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">How to Use SLSL, Case by Case</h1>

</div>


<p><strong>Last updated:</strong> 2018-07-22</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20180618)</code> </summary></p>
<p>The command <code>set.seed(20180618)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/tk382/SCNoisyClustering/tree/38c8f0f5208d139f9e62de8dcdeb2caafefc5b32" target="_blank">38c8f0f</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    R/.Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    analysis/pipeline/.Rhistory

Untracked files:
    Untracked:  ..gif
    Untracked:  .DS_Store
    Untracked:  R/.DS_Store
    Untracked:  R/LSLSL2.R
    Untracked:  R/makesparsekernel.R
    Untracked:  R/savesparsekernel.R
    Untracked:  analysis/.DS_Store
    Untracked:  analysis/bibliography.bib
    Untracked:  analysis/euclidean10SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean11SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean12SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean13SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean14SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean15SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean16SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean17SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean18SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean1SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean2SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean3SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean4SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean5SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean6SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean7SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean8SLSL_kernels_tmp.txt
    Untracked:  analysis/euclidean9SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson10SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson11SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson12SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson13SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson14SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson15SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson16SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson17SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson18SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson1SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson2SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson3SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson4SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson5SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson6SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson7SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson8SLSL_kernels_tmp.txt
    Untracked:  analysis/pearson9SLSL_kernels_tmp.txt
    Untracked:  analysis/pipeline/large_sets.pdf
    Untracked:  analysis/pipeline/temp_ari.txt
    Untracked:  analysis/pipeline/temp_time.txt
    Untracked:  analysis/spearman10SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman11SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman12SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman13SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman14SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman15SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman16SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman17SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman18SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman1SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman2SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman3SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman4SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman5SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman6SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman7SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman8SLSL_kernels_tmp.txt
    Untracked:  analysis/spearman9SLSL_kernels_tmp.txt
    Untracked:  analysis/tutorial_cache/
    Untracked:  analysis/writeup/cite.bib
    Untracked:  analysis/writeup/cite.log
    Untracked:  analysis/writeup/paper.aux
    Untracked:  analysis/writeup/paper.bbl
    Untracked:  analysis/writeup/paper.blg
    Untracked:  analysis/writeup/paper.log
    Untracked:  analysis/writeup/paper.out
    Untracked:  analysis/writeup/paper.synctex.gz
    Untracked:  analysis/writeup/paper.tex
    Untracked:  analysis/writeup/writeup.aux
    Untracked:  analysis/writeup/writeup.bbl
    Untracked:  analysis/writeup/writeup.blg
    Untracked:  analysis/writeup/writeup.dvi
    Untracked:  analysis/writeup/writeup.log
    Untracked:  analysis/writeup/writeup.out
    Untracked:  analysis/writeup/writeup.synctex.gz
    Untracked:  analysis/writeup/writeup.tex
    Untracked:  analysis/writeup/writeup2.aux
    Untracked:  analysis/writeup/writeup2.bbl
    Untracked:  analysis/writeup/writeup2.blg
    Untracked:  analysis/writeup/writeup2.log
    Untracked:  analysis/writeup/writeup2.out
    Untracked:  analysis/writeup/writeup2.pdf
    Untracked:  analysis/writeup/writeup2.synctex.gz
    Untracked:  analysis/writeup/writeup2.tex
    Untracked:  analysis/writeup/writeup3.aux
    Untracked:  analysis/writeup/writeup3.log
    Untracked:  analysis/writeup/writeup3.out
    Untracked:  analysis/writeup/writeup3.synctex.gz
    Untracked:  analysis/writeup/writeup3.tex
    Untracked:  data/unnecessary_in_building/

Unstaged changes:
    Deleted:    R/LSLSL.R
    Modified:   R/RcppExports.R
    Modified:   R/SLSL2.R
    Modified:   R/SLSL_ref.R
    Modified:   R/network.diffusion.R
    Deleted:    analysis/correcting_detection_rate/Buettner.Rmd
    Deleted:    analysis/correcting_detection_rate/Buettner.pdf
    Deleted:    analysis/correcting_detection_rate/Pollen.R
    Deleted:    analysis/correcting_detection_rate/Pollen.Rmd
    Deleted:    analysis/correcting_detection_rate/Pollen.pdf
    Deleted:    analysis/correcting_detection_rate/Usoskin.Rmd
    Deleted:    analysis/correcting_detection_rate/Usoskin.pdf
    Deleted:    analysis/correcting_detection_rate/Usoskin_files/figure-latex/loess residuals-1.pdf
    Deleted:    analysis/correcting_detection_rate/Usoskin_files/figure-latex/pca_plot-1.pdf
    Deleted:    analysis/correcting_detection_rate/Usoskin_files/figure-latex/quantile_plot-1.pdf
    Deleted:    analysis/correcting_detection_rate/Usoskin_files/figure-latex/quantile_plot_log-1.pdf
    Deleted:    analysis/correcting_detection_rate/Usoskin_files/figure-latex/rg residuals-1.pdf
    Deleted:    analysis/correcting_detection_rate/Zeisel.R
    Deleted:    analysis/correcting_detection_rate/detectionrate_alldata.R
    Deleted:    analysis/correcting_detection_rate/detectionrate_pbmc.R
    Deleted:    analysis/correcting_detection_rate/sc.xls
    Modified:   analysis/detection_rate_correction.Rmd
    Modified:   analysis/pipeline/large_sets.Rmd
    Modified:   analysis/pipeline/pbmc_pipeline.R
    Modified:   analysis/pipeline/small_good_sets.Rmd
    Modified:   analysis/pipeline/small_good_sets_result.txt
    Modified:   analysis/pipeline/small_good_sets_time.txt
    Modified:   analysis/writeup/.DS_Store
    Modified:   analysis/writeup/paper.pdf
    Modified:   data/.DS_Store
    Modified:   docs/figure/.DS_Store
    Modified:   src/RcppExports.cpp
    Modified:   src/RcppExports.o
    Modified:   src/SCNoisyClustering.cpp
    Modified:   src/SCNoisyClustering.o
    Modified:   src/SCNoisyClustering.so

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/tk382/SCNoisyClustering/a20c598fafe77f5576cf60dffa34901514f90900/docs/tutorial.html" target="_blank">a20c598</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-21
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/56bd21fa8e697249fca99d1b62d0faa25b0265a0/analysis/tutorial.Rmd" target="_blank">56bd21f</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-21
</td>
<td style="text-align:left;">
wflow_publish(c(“analysis/tutorial.Rmd”, “analysis/_site.yml“), republish =
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/tk382/SCNoisyClustering/5787a2efa1b4575c9591aac564d821a2de9c63b2/docs/tutorial.html" target="_blank">5787a2e</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-17
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/e84909db67550789c058170bfad6aa9a87580706/analysis/tutorial.Rmd" target="_blank">e84909d</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-17
</td>
<td style="text-align:left;">
wflow_publish(c(“analysis/tutorial.Rmd”, “analysis/detection_rate_correction.Rmd”,
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/tk382/SCNoisyClustering/a08d3beefc8907a60fb5c8a3def03153552faf71/docs/tutorial.html" target="_blank">a08d3be</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/0a9facc880c2cf3ade772eef6e887998df44a395/analysis/tutorial.Rmd" target="_blank">0a9facc</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
wflow_publish(“analysis/tutorial.Rmd”)
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/44d04548b9737baf48fb5b98a84cf6a38d8f4200/analysis/tutorial.Rmd" target="_blank">44d0454</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
fix Chu data error
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/tk382/SCNoisyClustering/44d04548b9737baf48fb5b98a84cf6a38d8f4200/docs/tutorial.html" target="_blank">44d0454</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
fix Chu data error
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/tk382/SCNoisyClustering/aa768aaa40c3995fffd5dead71b9d87149a6b038/docs/tutorial.html" target="_blank">aa768aa</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/tk382/SCNoisyClustering/2fb398fccf9c7b598fab0de707f8f12e36ab8471/docs/tutorial.html" target="_blank">2fb398f</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/3cf7635a67a03f17a8799d964848783d82e556f8/analysis/tutorial.Rmd" target="_blank">3cf7635</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
wflow_publish(“analysis/tutorial.Rmd”)
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/tk382/SCNoisyClustering/27cffd8d8348801fc744a988d139b7ad0e7540b3/docs/tutorial.html" target="_blank">27cffd8</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/de10fbe0966f4955f90180977e486585ed0f1306/analysis/tutorial.Rmd" target="_blank">de10fbe</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
wflow_publish(“analysis/tutorial.Rmd”)
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/tk382/SCNoisyClustering/1b226e63ee238b00a51ba0e1fea5e6fa70503d92/docs/tutorial.html" target="_blank">1b226e6</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/9e8c49b2665a941a25d83f8812e4862bbfd1a424/analysis/tutorial.Rmd" target="_blank">9e8c49b</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
fixed the null pointer error
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/tk382/SCNoisyClustering/9e8c49b2665a941a25d83f8812e4862bbfd1a424/docs/tutorial.html" target="_blank">9e8c49b</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
fixed the null pointer error
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/2d3d0b0949b80f79e1045ad3cd04788e188da000/analysis/tutorial.Rmd" target="_blank">2d3d0b0</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
<td style="text-align:left;">
solved mysterious null space error (removed importing r package)
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/f6c5b074e2aa03ff4cf9e37c2cea4a3b9fd6ec19/analysis/tutorial.Rmd" target="_blank">f6c5b07</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-11
</td>
<td style="text-align:left;">
add website
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/tk382/SCNoisyClustering/f6c5b074e2aa03ff4cf9e37c2cea4a3b9fd6ec19/docs/tutorial.html" target="_blank">f6c5b07</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-11
</td>
<td style="text-align:left;">
add website
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<hr />
<div id="load-the-slsl-library." class="section level1">
<h1>Load the SLSL library.</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(SCNoisyClustering)

<span class="kw">library</span>(data.table)
<span class="kw">library</span>(inline)
<span class="kw">library</span>(matrixStats)
<span class="kw">library</span>(quadprog)
<span class="kw">library</span>(irlba)
<span class="kw">library</span>(ggplot2)
<span class="kw">library</span>(dplyr)
<span class="kw">library</span>(reshape)
<span class="kw">library</span>(caret)
<span class="kw">library</span>(fossil)
<span class="kw">library</span>(pracma)
<span class="kw">library</span>(igraph)
<span class="kw">library</span>(Rtsne)
<span class="kw">library</span>(gplots)
<span class="kw">library</span>(broom)
<span class="kw">library</span>(abind)
<span class="kw">library</span>(stargazer)
<span class="kw">library</span>(scatterplot3d)
<span class="kw">library</span>(diceR)
<span class="kw">library</span>(parallel)

<span class="co"># set.seed(1)</span>
<span class="co"># R.utils::sourceDirectory(&quot;~/Dropbox/TaeProject/VariableError/SCNoisyClustering/R/&quot;, modifiedOnly=FALSE)</span>
<span class="co"># Rcpp::sourceCpp(&#39;~/Dropbox/TaeProject/VariableError/SCNoisyClustering/src/SCNoisyClustering.cpp&#39;)</span>

heat =<span class="st"> </span><span class="cf">function</span>(S){
  <span class="kw">ggplot</span>(<span class="kw">melt</span>(S), <span class="kw">aes</span>(<span class="dt">x=</span>X1, <span class="dt">y=</span>X2, <span class="dt">fill=</span>value)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_tile</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_fill_gradient2</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>, <span class="dt">hjust =</span> <span class="dv">1</span>)) <span class="op">+</span>
<span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>)
}</code></pre></div>
</div>
<div id="small-good-data-sets" class="section level1">
<h1>Small, good data sets</h1>
<p>First, prepare your data set. It should be a normalized count matrix in units such as TPM, CPM, RPKM, or FPKM. True label data is available for the sample data Yan et al. <span class="citation">(L. Yan et al. 2013)</span>, and we will evaluate the performance at the end as an example, but the label information is not expected for most cases.</p>
<div id="read-data" class="section level2">
<h2>Read Data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&#39;../data/Yan.rda&#39;</span>)
X         =<span class="st"> </span><span class="kw">as.matrix</span>(yan)
truelabel =<span class="st"> </span><span class="kw">as.character</span>(ann<span class="op">$</span>cell_type1)
numClust  =<span class="st"> </span><span class="dv">6</span>
<span class="kw">rm</span>(ann, yan)</code></pre></div>
<p>The data should look something like this.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">X[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]</code></pre></div>
<pre><code>         Oocyte..1.RPKM. Oocyte..2.RPKM. Oocyte..3.RPKM. Zygote..1.RPKM.
C9orf152             0.0             0.0             0.0             0.0
RPS11             1219.9          1021.1           931.6           875.4
ELMO2                7.0            12.2             9.3            52.1
CREB3L1              1.0             1.4             1.9             1.8</code></pre>
<p><span class="math inline">\(X\)</span> must be a matrix with dimension <span class="math inline">\(p \times n\)</span> where <span class="math inline">\(p\)</span> is the gene count and <span class="math inline">\(n\)</span> is the cell count. Each cell is located in each column, and each gene is located in each row. The sample data has the gene names as the row names, but this is not required unless you plan to use the reference panel. The column names are also not required because the package assumes the cell names contain no information about the cell subtypes.</p>
</div>
<div id="parameter-set-up" class="section level2">
<h2>Parameter Set-Up</h2>
<p>The SLSL wrapper function can be used like below. Below are the explanations of the parameters available for you to fine-tune.</p>
<p>First, you can specify the number of clusters. The default is “NA”, and SLSL will decide this for you based on the eigenvalues of the learned similarity matrix.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">numClust =<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(truelabel)) <span class="co">#4</span></code></pre></div>
<p>A knn parameter <span class="math inline">\(k\)</span> determines how many neighbors the algorithm uses to account for the local structure. The default is k = NA and algorithm will automatically assign it as max(10, number of cells / 20). This parameter is used in building the kernel matrices and in network diffusion.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">k =<span class="st"> </span><span class="ot">NA</span></code></pre></div>
<p>Then decide if you’d like to filter the genes. For each gene, SLSL counts the number of cells that have zero count, and if the proportion of such cells is higher than filter_p1 or lower than filter_p2, such gene is removed. This does not influence the clustering result much, unless filter_p1 is too low or filter_p2 is too high, and reduces the computational burden of the algorithm. The default choice of 0.9 and 0 reasonably selects the genes that do not hold much information for clustering and remove them.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">filter =<span class="st"> </span><span class="ot">TRUE</span> <span class="co">#or FALSE</span>
filter_p1 =<span class="st"> </span><span class="fl">0.9</span> <span class="co">#any value 0 and 1, but something between 0.85 - 0.95 is recommended</span>
filter_p2 =<span class="st"> </span><span class="dv">0</span> <span class="co">#any value between 0 and 1, but something less than 0.1 is recommended</span></code></pre></div>
<p>Hicks 2017 showed that a large amount of cell-to-cell variation comes from the detection rate, or the proportion of genes that have greater than zero count. For example, first principal component of the data is often correlated with the zero counts of each cell.</p>
<p>Hicks explains that log-transforming the datawith pseudo-counts introduces the bias because the mean counts are no longer the same across the cells.</p>
<p>The first singular vector is plotted below against the detection rate. It is shown that after centering each column of the log transformed matrix removes part of the correlation between the detection rate and the first singular vector. This correction influences the Euclidean distance matrix, and therefore f</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(irlba)
v =<span class="st"> </span><span class="kw">irlba</span>(<span class="kw">log</span>(X<span class="op">+</span><span class="dv">1</span>), <span class="dv">1</span>)<span class="op">$</span>v[,<span class="dv">1</span>]
det =<span class="st"> </span><span class="dv">1</span><span class="op">-</span><span class="kw">colSums</span>(X<span class="op">==</span><span class="dv">0</span>)<span class="op">/</span><span class="kw">nrow</span>(X)
<span class="kw">plot</span>(v <span class="op">~</span><span class="st"> </span>det)</code></pre></div>
<p><img src="figure/tutorial.Rmd/pca-1.png" width="336" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of pca-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a08d3beefc8907a60fb5c8a3def03153552faf71/docs/figure/tutorial.Rmd/pca-1.png" target="_blank">a08d3be</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/2fb398fccf9c7b598fab0de707f8f12e36ab8471/docs/figure/tutorial.Rmd/pca-1.png" target="_blank">2fb398f</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/1b226e63ee238b00a51ba0e1fea5e6fa70503d92/docs/figure/tutorial.Rmd/pca-1.png" target="_blank">1b226e6</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
</tbody>
</table>
<p></details></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">newX =<span class="st"> </span><span class="kw">scale</span>(<span class="kw">log</span>(X<span class="op">+</span><span class="dv">1</span>), <span class="dt">scale=</span>F)
newv =<span class="st"> </span><span class="kw">irlba</span>(newX, <span class="dv">1</span>)<span class="op">$</span>v[,<span class="dv">1</span>]
<span class="kw">plot</span>(newv<span class="op">~</span>det)</code></pre></div>
<p><img src="figure/tutorial.Rmd/pca-2.png" width="336" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of pca-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a08d3beefc8907a60fb5c8a3def03153552faf71/docs/figure/tutorial.Rmd/pca-2.png" target="_blank">a08d3be</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/2fb398fccf9c7b598fab0de707f8f12e36ab8471/docs/figure/tutorial.Rmd/pca-2.png" target="_blank">2fb398f</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/1b226e63ee238b00a51ba0e1fea5e6fa70503d92/docs/figure/tutorial.Rmd/pca-2.png" target="_blank">1b226e6</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>One way to correct this is to regress out the zero counts of each cell. However, as shown above, some data sets have non-linear relationship with the zero counts, and sometimes it’s better to transform the data accordingly.</p>
<p>When correct_detection_rate parameter is set to TRUE, the residuals after regressing out the zero counts are used instead of the log of counts. The default is FALSE, and we expect the users to inspect the data like above before setting this to TRUE.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">correct_detection_rate =<span class="st"> </span><span class="ot">TRUE</span> <span class="co">#default : FALSE</span></code></pre></div>
<p>Next argument is kernel_type. This determines the method to meaure the cell to cell distance. SLSL algorithm combines information from many distance matrices computed from different kernel parameters, where the default setting uses 18 different combinations of the kernel parameters. When kernel_type is specified as one of “spearman”, “pearson”, or “euclidean”, SLSL will learn similarity matrix based on those 18 kernel matrices using the specified distance masure. When kernel_type is specified as “combined” (default setting), SLSL will compute three distance matrices using all three distance measures and <span class="math inline">\(18 \times 3\)</span> kernel matrices to learn the similarity matrix. In other words, kernel_type = “combined” requires more memory and computational power, but it will reach higher accuracy by accounting for complex cell to cell variability.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">kernel_type =<span class="st"> &quot;combined&quot;</span> <span class="co">#or spearman or pearson or euclidean</span></code></pre></div>
<p>The user can also specify the kernel parameters. Denoting the distance from cell <span class="math inline">\(i\)</span> and cell <span class="math inline">\(j\)</span> as <span class="math inline">\(D_{ij}\)</span>, <span class="math inline">\(K_{ij}\)</span> given kernel parameters <span class="math inline">\(k\)</span> and <span class="math inline">\(\sigma\)</span> is <span class="math display">\[K_{ij} = \frac{1}{\sqrt{2\pi}\epsilon_{ij}} exp \left(-\frac{D_{ij}}{2\epsilon^2_{ij}}\right)\]</span> where <span class="math display">\[\epsilon_{ij} = \frac{(\mu_i + \mu_j)\sigma}{2}\]</span></p>
<p><span class="math display">\[\mu_{i} = \frac{\sum_{j \in KNN(i, k)} \|x_i - x_j\|^2_2}{k}\]</span> <span class="math inline">\(KNN(x_i, k)\)</span> is the <span class="math inline">\(k\)</span> nearest neighbors of cell <span class="math inline">\(i\)</span>, and <span class="math inline">\(x_i\)</span> is the gene expression level profile of cell <span class="math inline">\(i\)</span>. Therefore, the kernel parameters <span class="math inline">\(k\)</span> determines how many neighbors of each cell to use to account for the local structure, and another parameter <span class="math inline">\(\sigma\)</span> determines how to scale the distance to account for the non-linearity.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">klist =<span class="st"> </span><span class="kw">seq</span>(<span class="dv">15</span>, <span class="dv">25</span>, <span class="dt">by=</span><span class="dv">5</span>)
sigmalist =<span class="st"> </span><span class="kw">seq</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dt">by=</span><span class="fl">0.2</span>)</code></pre></div>
<p><span class="math inline">\(\tau\)</span> and <span class="math inline">\(\gamma\)</span> are the penalty parameters for similarity learning. SLSL combines multiple kernel matrices by giving each of them different weights to reach maximum sparsity (<span class="math inline">\(\tau\)</span>). Some past works showed that imposing penalty <span class="math inline">\(gamma\)</span> on the Frobenius norm of the learned similarity matrix. This prevents the similarity matrix from being too close to the identity matrix (assigning each cell to a separate cluster). However, empirically, removing this penalty and setting <span class="math inline">\(\gamma = 0\)</span> performed well. <span class="math inline">\(\tau\)</span> is set to 5 as a default, and as long as it’s not too small (we recommend <span class="math inline">\(\tau &gt; 1\)</span>), the algorithm works well.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tau =<span class="st"> </span><span class="dv">5</span>
gamma =<span class="st"> </span><span class="dv">0</span></code></pre></div>
<p>Lastly, verbose should be set to TRUE if the user would like to see the progress of the algorithm. measuretime should be set to TRUE if the user would like to see how long each step of the algorithm takes. The default settings are FALSE for both parameters, but here, we set them TRUE to see what they do.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">verbose =<span class="st"> </span><span class="ot">TRUE</span>
measuretime =<span class="st"> </span><span class="ot">TRUE</span></code></pre></div>
</div>
<div id="run-the-function" class="section level2">
<h2>Run the Function</h2>
<p>Using the prespecified parameters above, run the SLSL function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">out =<span class="st"> </span><span class="kw">SLSL2</span>(<span class="dt">X                     =</span> <span class="kw">as.matrix</span>(X),
           <span class="dt">numClust               =</span> numClust,
           <span class="dt">k                      =</span> k,
           <span class="dt">filter                 =</span> filter,
           <span class="dt">filter_p1              =</span> filter_p1,
           <span class="dt">filter_p2              =</span> filter_p2,
           <span class="dt">correct_detection_rate =</span> correct_detection_rate,
           <span class="dt">kernel_type            =</span> kernel_type,
           <span class="dt">klist                  =</span> klist,
           <span class="dt">sigmalist              =</span> sigmalist,
           <span class="dt">tau                    =</span> tau,
           <span class="dt">gamma                  =</span> gamma,
           <span class="dt">verbose                =</span> verbose,
           <span class="dt">measuretime            =</span> measuretime
           )</code></pre></div>
<pre><code>[1] &quot;constructing kernel..&quot;
[1] &quot;optimizing..&quot;
[1] &quot;network diffusion..&quot;
[1] &quot;dimension reduction..&quot;
gene filter &amp; numClust 0.39 seconds 
constructing kernel 2.5 seconds 
get S 0.11 seconds 
network diffusion 0.02 seconds
tsne 0.05 seconds</code></pre>
<p>Or, you can also use default parameters, and this will work as well.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">out =<span class="st"> </span><span class="kw">SLSL2</span>(X)</code></pre></div>
</div>
<div id="analyze-the-result" class="section level2">
<h2>Analyze the Result</h2>
<p>First, let’s take a look at the heatmap of the similarity matrix. Note that the cells have been arranged according to the true groupings. The result seems pretty good, except the two cells around 50 that have clearly been misclassified.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ind =<span class="st"> </span><span class="kw">sort</span>(out<span class="op">$</span>result, <span class="dt">index.return=</span><span class="ot">TRUE</span>)<span class="op">$</span>ix
<span class="kw">heat</span>(out<span class="op">$</span>S[ind, ind])</code></pre></div>
<p><img src="figure/tutorial.Rmd/similarity_matrix-1.png" width="336" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of similarity_matrix-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a20c598fafe77f5576cf60dffa34901514f90900/docs/figure/tutorial.Rmd/similarity_matrix-1.png" target="_blank">a20c598</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-21
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/5787a2efa1b4575c9591aac564d821a2de9c63b2/docs/figure/tutorial.Rmd/similarity_matrix-1.png" target="_blank">5787a2e</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-17
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a08d3beefc8907a60fb5c8a3def03153552faf71/docs/figure/tutorial.Rmd/similarity_matrix-1.png" target="_blank">a08d3be</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/2fb398fccf9c7b598fab0de707f8f12e36ab8471/docs/figure/tutorial.Rmd/similarity_matrix-1.png" target="_blank">2fb398f</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/27cffd8d8348801fc744a988d139b7ad0e7540b3/docs/figure/tutorial.Rmd/similarity_matrix-1.png" target="_blank">27cffd8</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>Below is the comparison with the true label. This is not applicable when true labels are not available.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">adj.rand.index</span>(out<span class="op">$</span>result, <span class="kw">as.numeric</span>(<span class="kw">as.factor</span>(truelabel)))</code></pre></div>
<pre><code>[1] 0.8954618</code></pre>
<p>Now let’s visualize the result through the dimension reduction result. The true label information allows us to analyze the errors from the algorithm. For example, the zygotes and 2-cells were clustered into one group, while there are two outliers in the 16 cells that are close to 8 cells.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tsne =<span class="st"> </span>out<span class="op">$</span>tsne
df =<span class="st"> </span><span class="kw">as.data.frame</span>(tsne)
df<span class="op">$</span>truelabel =<span class="st"> </span><span class="kw">as.factor</span>(truelabel)
df<span class="op">$</span>result =<span class="st"> </span><span class="kw">as.factor</span>(out<span class="op">$</span>result)

<span class="co">#ggplot(df, aes(x=V3, y=V2, col=truelabel)) + geom_point()</span>
<span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">x=</span>V3, <span class="dt">y=</span>V2, <span class="dt">col=</span>result)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_point</span>()</code></pre></div>
<p><img src="figure/tutorial.Rmd/dimred-1.png" width="336" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of dimred-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a20c598fafe77f5576cf60dffa34901514f90900/docs/figure/tutorial.Rmd/dimred-1.png" target="_blank">a20c598</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-21
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/5787a2efa1b4575c9591aac564d821a2de9c63b2/docs/figure/tutorial.Rmd/dimred-1.png" target="_blank">5787a2e</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-17
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a08d3beefc8907a60fb5c8a3def03153552faf71/docs/figure/tutorial.Rmd/dimred-1.png" target="_blank">a08d3be</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/2fb398fccf9c7b598fab0de707f8f12e36ab8471/docs/figure/tutorial.Rmd/dimred-1.png" target="_blank">2fb398f</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/1b226e63ee238b00a51ba0e1fea5e6fa70503d92/docs/figure/tutorial.Rmd/dimred-1.png" target="_blank">1b226e6</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
</tbody>
</table>
<p></details></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#scatterplot3d(x=df$V2, y=df$V3, z=df$V4, color = as.numeric(as.factor(truelabel)))</span>
<span class="kw">scatterplot3d</span>(<span class="dt">x=</span>df<span class="op">$</span>V2, <span class="dt">y=</span>df<span class="op">$</span>V3, <span class="dt">z=</span>df<span class="op">$</span>V4, <span class="dt">color =</span> df<span class="op">$</span>result)</code></pre></div>
<p><img src="figure/tutorial.Rmd/dimred-2.png" width="336" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of dimred-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a20c598fafe77f5576cf60dffa34901514f90900/docs/figure/tutorial.Rmd/dimred-2.png" target="_blank">a20c598</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-21
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/5787a2efa1b4575c9591aac564d821a2de9c63b2/docs/figure/tutorial.Rmd/dimred-2.png" target="_blank">5787a2e</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-17
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a08d3beefc8907a60fb5c8a3def03153552faf71/docs/figure/tutorial.Rmd/dimred-2.png" target="_blank">a08d3be</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/2fb398fccf9c7b598fab0de707f8f12e36ab8471/docs/figure/tutorial.Rmd/dimred-2.png" target="_blank">2fb398f</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/1b226e63ee238b00a51ba0e1fea5e6fa70503d92/docs/figure/tutorial.Rmd/dimred-2.png" target="_blank">1b226e6</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
</tbody>
</table>
<p></details></p>
</div>
</div>
<div id="using-reference-panel" class="section level1">
<h1>Using Reference Panel</h1>
<p>The SLSL function also has a hidden argument “ref”. This allows the users to utilize the reference panel data sets <span class="citation">(H. Li et al. 2017)</span>. Two panels are available: cell types and tissue types. One can simply add this parameter in SLSL function.</p>
<div id="demonstration1-10x-data-of-pbmc-cells" class="section level2">
<h2>Demonstration1 : 10X data of PBMC cells</h2>
<p>To demonstrate the use of the reference panel, we use the 10X data of PBMC cells from [Zheng2017massively]. They provide the reference set of 11 type of pure cell type populations, and we use the file for demonstration. Due to the size, these data sets are not included in the package, and they can be found here: <a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets" class="uri">https://support.10xgenomics.com/single-cell-gene-expression/datasets</a> and <a href="https://github.com/10XGenomics/single-cell-3prime-paper" class="uri">https://github.com/10XGenomics/single-cell-3prime-paper</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">X =<span class="st"> </span><span class="kw">readMM</span>(<span class="st">&#39;../data/unnecessary_in_building/pbmc3k/matrix.mtx&#39;</span>)
X =<span class="st"> </span><span class="kw">as.matrix</span>(X)
genes =<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;../data/unnecessary_in_building/pbmc3k/genes.tsv&#39;</span>)
<span class="kw">rownames</span>(X) =<span class="st"> </span>genes<span class="op">$</span>V2

ref_file =<span class="st"> </span><span class="kw">readRDS</span>(<span class="st">&#39;../data/unnecessary_in_building/pbmc3k/all_pure_select_11types.rds&#39;</span>)
ref =<span class="st"> </span><span class="kw">t</span>(ref_file<span class="op">$</span>pure_avg)
<span class="kw">rownames</span>(ref) =<span class="st"> </span>genes<span class="op">$</span>V2[ref_file<span class="op">$</span>pure_use_genes]
<span class="kw">colnames</span>(ref) =<span class="st"> </span>ref_file<span class="op">$</span>pure_id
ref =<span class="st"> </span><span class="kw">scale</span>(ref)</code></pre></div>
<p>The two most important arguments are the data matrix “X” and reference panel “ref”. We assume X is count matrix, while reference panel is already log transformed and normalized. Another parameter the user can choose is “knn”. Empirical results showed that, after taking the projection of the data matrix onto the reference panel, removing low correlations improve the signal strength.</p>
<p>Note that 10X data has very low detection rate. (number of genes detected as greater than 0 for each cell). The histogram below shows that only around 2% of the genes are detected for most cells.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(<span class="kw">colSums</span>(X<span class="op">!=</span><span class="dv">0</span>) <span class="op">/</span><span class="st"> </span><span class="kw">nrow</span>(X), <span class="dt">breaks=</span><span class="dv">50</span>,
     <span class="dt">xlab =</span> <span class="st">&#39;detection rate per cell&#39;</span>, <span class="dt">ylab =</span> <span class="st">&#39;&#39;</span>)</code></pre></div>
<p><img src="figure/tutorial.Rmd/10Xdet-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of 10Xdet-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/5787a2efa1b4575c9591aac564d821a2de9c63b2/docs/figure/tutorial.Rmd/10Xdet-1.png" target="_blank">5787a2e</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-17
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>This data is from 11 different PBMC cell types, and yet the dimension reduction doesn’t show any particular patterns.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tsne =<span class="st"> </span><span class="kw">Rtsne</span>(<span class="kw">t</span>(<span class="kw">log</span>(X<span class="op">+</span><span class="dv">1</span>)), <span class="dt">perplexity=</span><span class="dv">10</span>)</code></pre></div>
<p>When we use the reference panel, the clustering becomes much easier.</p>
<p>A couple things to note. Note that the projection matrix will be very low rank. With this particular data set, there are 11 reference cells with 2,700 cells to cluster, so the similarity matrix will be a 2,700 2700 matrix with rank 11. That means, even when the projection matrix is filled with completely random numbers, the similarity matrix will have a certain structure with rank less than or equal to 11. Therefore, the result may be over-confident about the data structure. Another thing to note is that, since there are only few features for each cell in projection matrix, when we use Spearman correlation to measure similarity, there will be cells with correlation 1, causing numerical instability. Therefore, we strongly recommend that users use Euclidean distance to build a kernel.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pbmc_ref =<span class="st"> </span><span class="kw">SLSL_ref</span>(<span class="dt">X =</span> <span class="kw">as.matrix</span>(X),
                    <span class="dt">filter =</span> <span class="ot">TRUE</span>,
                    <span class="dt">filter_p1 =</span> <span class="fl">0.9</span>,
                    <span class="dt">filter_p2 =</span> <span class="fl">0.1</span>,
                    <span class="dt">kernel_type =</span> <span class="st">&quot;euclidean&quot;</span>,
                    <span class="dt">numClust =</span> <span class="dv">11</span>,
                    <span class="dt">ref =</span> ref, 
                    <span class="dt">verbose=</span>T)</code></pre></div>
<pre><code>[1] &quot;data cleaning..&quot;
[1] &quot;computing projection..&quot;
[1] &quot;ref with smaller version..&quot;
[1] &quot;constructing kernel..&quot;
[1] &quot;optimizing..&quot;
[1] &quot;network diffusion..&quot;
[1] &quot;dimension reduction..&quot;</code></pre>
<p>Sort the cells and visualize the projection matrix.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">heat</span>(pbmc_ref<span class="op">$</span>projection[,<span class="kw">sort</span>(pbmc_ref<span class="op">$</span>result<span class="op">$</span>result, <span class="dt">index.return=</span>T)<span class="op">$</span>ix])</code></pre></div>
<p><img src="figure/tutorial.Rmd/pbmc_analyze-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of pbmc_analyze-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a20c598fafe77f5576cf60dffa34901514f90900/docs/figure/tutorial.Rmd/pbmc_analyze-1.png" target="_blank">a20c598</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-21
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/5787a2efa1b4575c9591aac564d821a2de9c63b2/docs/figure/tutorial.Rmd/pbmc_analyze-1.png" target="_blank">5787a2e</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-17
</td>
</tr>
</tbody>
</table>
<p></details></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(tsne<span class="op">$</span>Y, <span class="dt">col =</span> pbmc_ref<span class="op">$</span>result<span class="op">$</span>result)</code></pre></div>
<p><img src="figure/tutorial.Rmd/pbmc_analyze-2.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of pbmc_analyze-2.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a20c598fafe77f5576cf60dffa34901514f90900/docs/figure/tutorial.Rmd/pbmc_analyze-2.png" target="_blank">a20c598</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-21
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/5787a2efa1b4575c9591aac564d821a2de9c63b2/docs/figure/tutorial.Rmd/pbmc_analyze-2.png" target="_blank">5787a2e</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-17
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>We also provide two reference sets of human cell types and tissue types. Below is a demonstration of cell type panel.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sysdata =<span class="st"> </span>SCNoisyClustering<span class="op">::</span>reference_panel
ref =<span class="st"> </span>reference_panel<span class="op">$</span>cell

<span class="kw">load</span>(<span class="st">&#39;../data/unnecessary_in_building/7_Chu_celltype.RData&#39;</span>)
X =<span class="st"> </span>Chu_celltype<span class="op">$</span>X
chu_ref =<span class="st"> </span><span class="kw">SLSL_ref</span>(<span class="dt">X =</span> <span class="kw">as.matrix</span>(X),
                   <span class="dt">ref =</span> ref, 
                   <span class="dt">numClust =</span> <span class="dv">7</span>,
                   <span class="dt">verbose =</span> T)</code></pre></div>
<pre><code>[1] &quot;data cleaning..&quot;
[1] &quot;computing projection..&quot;
[1] &quot;ref with smaller version..&quot;
[1] &quot;constructing kernel..&quot;
[1] &quot;optimizing..&quot;
[1] &quot;network diffusion..&quot;
[1] &quot;dimension reduction..&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">heat</span>(chu_ref<span class="op">$</span>projection[,<span class="kw">sort</span>(chu_ref<span class="op">$</span>result<span class="op">$</span>result, <span class="dt">index.return=</span>T)<span class="op">$</span>ix])</code></pre></div>
<p><img src="figure/tutorial.Rmd/visualize_chu_ref_projection-1.png" width="672" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of visualize_chu_ref_projection-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a20c598fafe77f5576cf60dffa34901514f90900/docs/figure/tutorial.Rmd/visualize_chu_ref_projection-1.png" target="_blank">a20c598</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-21
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>One can observe that in the cell atlas panel, ESC (embryonic stem cells) tend to show high correlation, which is as expected. Now, take a look at the heatmap of the similarity matrix of the projection matrix, and evaluate the clustering result.</p>
<p>Since using reference panels often throws away a lot of information, users should be careful when to use panels and which panel to use. The data sets should be from human cells, and they should have enough cellular heterogeneity. For example, Yan data are from different stages of human embryonic cells, and once projected on to the reference panels, all the cells belonged to one type : oocytes. The Chu data above also shows a much better performance when clustered without the reference panel.</p>
</div>
</div>
<div id="large-sets-cells-3000" class="section level1">
<h1>Large Sets (cells &gt; 3000)</h1>
<p>There is a separate function for data sets that have more than 3,000 cells : LSLSL(Large Similarity Learning with Scaled Lasso).</p>
<p>The input data matrix is the same as regular SLSL: a count matrix with cells in columns and genes in rows. LSLSL assumes that the number of columns exceed 1000. We use Zeisel data <span class="citation">(Zeisel et al. 2015)</span></p>
<div id="read-data-1" class="section level2">
<h2>Read Data</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&#39;../data/unnecessary_in_building/Zeisel.Rdata&#39;</span>)
X =<span class="st"> </span>Zeisel<span class="op">$</span>X
truelabel =<span class="st"> </span>Zeisel<span class="op">$</span>label</code></pre></div>
<p>Note that Zeisel is a Drop-seq data, and coverage is very low. The histogram of the detection rate is below.</p>
</div>
<div id="parameter-set-up-1" class="section level2">
<h2>Parameter Set-Up</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">detection_rate =<span class="st"> </span><span class="dv">1</span><span class="op">-</span><span class="kw">colSums</span>(X<span class="op">==</span><span class="dv">0</span>)<span class="op">/</span><span class="kw">nrow</span>(X)
<span class="kw">hist</span>(detection_rate, <span class="dt">breaks =</span> <span class="dv">20</span>)</code></pre></div>
<p><img src="figure/tutorial.Rmd/zeisel_coverage-1.png" width="336" style="display: block; margin: auto;" /></p>
<details> <summary><em>Expand here to see past versions of zeisel_coverage-1.png:</em></summary>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/a08d3beefc8907a60fb5c8a3def03153552faf71/docs/figure/tutorial.Rmd/zeisel_coverage-1.png" target="_blank">a08d3be</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://github.com/tk382/SCNoisyClustering/blob/2fb398fccf9c7b598fab0de707f8f12e36ab8471/docs/figure/tutorial.Rmd/zeisel_coverage-1.png" target="_blank">2fb398f</a>
</td>
<td style="text-align:left;">
tk382
</td>
<td style="text-align:left;">
2018-07-12
</td>
</tr>
</tbody>
</table>
<p></details></p>
<p>Since we have the true label data, we will use the true number of clusters,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">numClust =<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(truelabel)) <span class="co">#default is NA</span></code></pre></div>
<p>For the kernel_type, we will use “combined” as before.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">kernel_type =<span class="st"> &quot;combined&quot;</span> <span class="co">#one of &quot;spearman&quot;, &quot;pearson&quot;, or &quot;euclidean&quot;</span></code></pre></div>
<p>LSLSL uses “parallel” package in CRAN for parallel computing. Users can specify the number of cores to use. The default is NA, and LSLSL will detect the number of cores of your laptop and use everything except 1. Here, we specify it as 2.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">core =<span class="st"> </span><span class="dv">3</span> <span class="co">#default is NA</span></code></pre></div>
<p>The shuffle argument decides if LSLSL should shuffle the order of the cells. LSLSL divides the cells into groups of approximately the same size, and it works the best if each group contains all subtypes of the cells. Unless the user carefully designed the ordering of the cells so that they are well mixed, we recommend to set shuffle as TRUE.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">shuffle =<span class="st"> </span><span class="ot">TRUE</span> <span class="co">#default is TRUE</span></code></pre></div>
<p>Now run the algorithm with different methods.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res =<span class="st"> </span><span class="kw">LSLSL2</span>(<span class="dt">X              =</span> X,
            <span class="dt">numClust       =</span> numClust,
            <span class="dt">kernel_type    =</span> kernel_type,
            <span class="dt">core           =</span> core,
            <span class="dt">shuffle        =</span> shuffle,
            <span class="dt">cluster_method =</span> <span class="st">&quot;CSPA&quot;</span>,
            <span class="dt">verbose        =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>[1] &quot;building pearson correlation matrix..&quot;
[1] &quot;      now saving following kernels&quot;
[1] &quot;           pearson1SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson2SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson3SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson4SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson5SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson6SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson7SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson8SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson9SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson10SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson11SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson12SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson13SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson14SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson15SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson16SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson17SLSL_kernels_tmp.txt&quot;
[1] &quot;           pearson18SLSL_kernels_tmp.txt&quot;
[1] &quot;building euclidean distance matrix..&quot;
[1] &quot;      now saving following kernels&quot;
[1] &quot;           euclidean1SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean2SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean3SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean4SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean5SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean6SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean7SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean8SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean9SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean10SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean11SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean12SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean13SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean14SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean15SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean16SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean17SLSL_kernels_tmp.txt&quot;
[1] &quot;           euclidean18SLSL_kernels_tmp.txt&quot;
[1] &quot;building euclidean distance matrix..&quot;
[1] &quot;      now saving following kernels&quot;
[1] &quot;           spearman1SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman2SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman3SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman4SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman5SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman6SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman7SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman8SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman9SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman10SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman11SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman12SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman13SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman14SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman15SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman16SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman17SLSL_kernels_tmp.txt&quot;
[1] &quot;           spearman18SLSL_kernels_tmp.txt&quot;
[1] &quot;working on the following subsets:&quot;
[1] &quot;consensus clustering: CSPA..&quot;</code></pre>
</div>
<div id="analyze-the-result-1" class="section level2">
<h2>Analyze the Result</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">adj.rand.index</span>(truelabel, res<span class="op">$</span>result)</code></pre></div>
<pre><code>[1] 0.640202</code></pre>
<p>Let’s try all the available consensus clustering methods in LSLSL function. The result is stored, so there is no need to run it again. The following chunk arranges the data set into a 4 dimensional array for the specification of package “diceR.” <span class="citation">(Chiu and Talhouk 2018)</span></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># construct an array called &quot;out&quot; for specification in diceR package.</span>
final =<span class="st"> </span>res<span class="op">$</span>array
out =<span class="st"> </span><span class="kw">array</span>(<span class="dv">0</span>, <span class="dt">dim=</span><span class="kw">c</span>(<span class="kw">nrow</span>(final), <span class="kw">ncol</span>(final), <span class="dv">1</span>, <span class="dv">1</span>))
out[,,<span class="dv">1</span>,<span class="dv">1</span>] =<span class="st"> </span>final
k =<span class="st"> </span><span class="kw">length</span>(<span class="kw">unique</span>(truelabel))

<span class="kw">dimnames</span>(out)[[<span class="dv">1</span>]] =<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&#39;R&#39;</span>,<span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(final)) <span class="co">#random row names</span>
<span class="kw">dimnames</span>(out)[[<span class="dv">2</span>]] =<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&#39;C&#39;</span>,<span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(final)) <span class="co">#random column names</span>
<span class="kw">dimnames</span>(out)[[<span class="dv">3</span>]] =<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;k&quot;</span>)
<span class="kw">dimnames</span>(out)[[<span class="dv">4</span>]] =<span class="st"> </span><span class="kw">as.character</span>(k) <span class="co">#This must be the number of clusters</span></code></pre></div>
<p>You must impute the array in order to use other consensus clustering methods. The “impute_missing” requires the raw data matrix (make sure it is transposed), and here we use the filtered X. This is appropriate especially because this is a Drop-seq data where most genes hold little information. The three methods are “k_modes”, “majority_voting”, and “LCE”. For the guidance of which method to use, refer to <span class="citation">(Chiu and Talhouk 2018)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">X =<span class="st"> </span><span class="kw">log</span>(<span class="kw">genefilter</span>(X)<span class="op">+</span><span class="dv">1</span>)
E =<span class="st"> </span><span class="kw">impute_missing</span>(out, <span class="kw">t</span>(X), <span class="dv">9</span>)
kmodes =<span class="st"> </span><span class="kw">k_modes</span>(E, <span class="dv">9</span>)
majvote =<span class="st"> </span><span class="kw">majority_voting</span>(E, <span class="dv">9</span>)
lce =<span class="st"> </span><span class="kw">LCE</span>(E, <span class="dv">9</span>)
<span class="kw">adj.rand.index</span>(kmodes, truelabel)
<span class="kw">adj.rand.index</span>(majvote, truelabel)
<span class="kw">adj.rand.index</span>(lce, truelabel) </code></pre></div>
<div id="refs">
<div id="ref-diceR">
<p>Chiu, Derek, and Aline Talhouk. 2018. <em>DiceR: Diverse Cluster Ensemble in R</em>. <a href="https://CRAN.R-project.org/package=diceR" class="uri">https://CRAN.R-project.org/package=diceR</a>.</p>
</div>
<div id="ref-li2017reference">
<p>Li, Huipeng, Elise T Courtois, Debarka Sengupta, Yuliana Tan, Kok Hao Chen, Jolene Jie Lin Goh, Say Li Kong, et al. 2017. “Reference Component Analysis of Single-Cell Transcriptomes Elucidates Cellular Heterogeneity in Human Colorectal Tumors.” <em>Nature Genetics</em> 49 (5). Nature Publishing Group: 708.</p>
</div>
<div id="ref-yan2013single">
<p>Yan, Liying, Mingyu Yang, Hongshan Guo, Lu Yang, Jun Wu, Rong Li, Ping Liu, et al. 2013. “Single-Cell Rna-Seq Profiling of Human Preimplantation Embryos and Embryonic Stem Cells.” <em>Nature Structural and Molecular Biology</em> 20 (9). Nature Publishing Group: 1131.</p>
</div>
<div id="ref-zeisel2015cell">
<p>Zeisel, Amit, Ana B Muñoz-Manchado, Simone Codeluppi, Peter Lönnerberg, Gioele La Manno, Anna Juréus, Sueli Marques, et al. 2015. “Cell Types in the Mouse Cortex and Hippocampus Revealed by Single-Cell Rna-Seq.” <em>Science</em> 347 (6226). American Association for the Advancement of Science: 1138–42.</p>
</div>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()</code></pre></div>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.5

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] diceR_0.5.1             scatterplot3d_0.3-41   
 [3] stargazer_5.2.2         abind_1.4-5            
 [5] broom_0.4.5             gplots_3.0.1           
 [7] Rtsne_0.13              igraph_1.2.1           
 [9] pracma_2.1.4            fossil_0.3.7           
[11] shapefiles_0.7          foreign_0.8-70         
[13] maps_3.3.0              sp_1.2-7               
[15] caret_6.0-80            lattice_0.20-35        
[17] reshape_0.8.7           dplyr_0.7.6            
[19] ggplot2_3.0.0           irlba_2.3.2            
[21] Matrix_1.2-14           quadprog_1.5-5         
[23] matrixStats_0.53.1      inline_0.3.15          
[25] data.table_1.10.4-3     SCNoisyClustering_0.1.0

loaded via a namespace (and not attached):
 [1] nlme_3.1-137       bitops_1.0-6       lubridate_1.7.4   
 [4] dimRed_0.1.0       rprojroot_1.3-2    tools_3.5.1       
 [7] backports_1.1.2    R6_2.2.2           KernSmooth_2.23-15
[10] rpart_4.1-13       lazyeval_0.2.1     colorspace_1.3-2  
[13] nnet_7.3-12        withr_2.1.2        tidyselect_0.2.4  
[16] mnormt_1.5-5       compiler_3.5.1     git2r_0.21.0      
[19] labeling_0.3       caTools_1.17.1     scales_0.5.0      
[22] sfsmisc_1.1-2      DEoptimR_1.0-8     psych_1.8.3.3     
[25] robustbase_0.93-0  stringr_1.3.0      digest_0.6.15     
[28] rmarkdown_1.9      R.utils_2.6.0      pkgconfig_2.0.1   
[31] htmltools_0.3.6    rlang_0.2.0        ddalpha_1.3.2     
[34] bindr_0.1.1        mclust_5.4         gtools_3.5.0      
[37] ModelMetrics_1.1.0 R.oo_1.22.0        magrittr_1.5      
[40] Rcpp_0.12.16       munsell_0.4.3      R.methodsS3_1.7.1 
[43] stringi_1.1.7      whisker_0.3-2      yaml_2.1.19       
[46] MASS_7.3-50        plyr_1.8.4         recipes_0.1.2     
[49] grid_3.5.1         gdata_2.18.0       splines_3.5.1     
[52] knitr_1.20         pillar_1.2.2       reshape2_1.4.3    
[55] codetools_0.2-15   stats4_3.5.1       CVST_0.2-1        
[58] magic_1.5-8        glue_1.2.0         evaluate_0.10.1   
[61] foreach_1.4.4      gtable_0.2.0       purrr_0.2.4       
[64] tidyr_0.8.0        kernlab_0.9-25     assertthat_0.2.0  
[67] DRR_0.0.3          gower_0.1.2        prodlim_2018.04.18
[70] class_7.3-14       survival_2.42-3    geometry_0.3-6    
[73] timeDate_3043.102  RcppRoll_0.2.2     tibble_1.4.2      
[76] iterators_1.0.9    workflowr_1.1.1    bindrcpp_0.2.2    
[79] lava_1.6.1         ipred_0.9-6       </code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
